extra/fuegogin/adaptor.go

```diff
@@ -0,0 +1,78 @@
+package fuegogin
+
+import (
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/gin-gonic/gin"
+
+	"github.com/go-fuego/fuego"
+)
+
+func Get[T, B any](
+	s *fuego.OpenAPI,
+	e *gin.Engine,
+	path string,
+	handler func(c *ContextWithBody[B]) (T, error),
+	options ...func(*fuego.BaseRoute),
+) *fuego.Route[T, B] {
+	return Handle(s, e, "GET", path, handler, options...)
+}
+
+func Post[T, B any](
+	s *fuego.OpenAPI,
+	e *gin.Engine,
+	path string,
+	handler func(c *ContextWithBody[B]) (T, error),
+	options ...func(*fuego.BaseRoute),
+) *fuego.Route[T, B] {
+	return Handle(s, e, "POST", path, handler, options...)
+}
+
+func Handle[T, B any](
+	openapi *fuego.OpenAPI,
+	e *gin.Engine,
+	method,
+	path string,
+	handler func(c *ContextWithBody[B]) (T, error),
+	options ...func(*fuego.BaseRoute),
+) *fuego.Route[T, B] {
+	route := &fuego.Route[T, B]{
+		BaseRoute: fuego.BaseRoute{
+			Method:    method,
+			Path:      path,
+			Params:    make(map[string]fuego.OpenAPIParam),
+			FullName:  fuego.FuncName(handler),
+			Operation: openapi3.NewOperation(),
+			OpenAPI:   openapi,
+		},
+	}
+
+	for _, o := range options {
+		o(&route.BaseRoute)
+	}
+
+	route.BaseRoute.GenerateDefaultDescription()
+
+	e.Handle(method, path, func(c *gin.Context) {
+		context := &ContextWithBody[B]{
+
+			ginCtx: c,
+		}
+
+		ans, err := handler(context)
+		if err != nil {
+			c.Error(err)
+			return
+		}
+
+		if c.Request.Header.Get("Accept") == "application/xml" {
+			c.XML(200, ans)
+			return
+		}
+
+		c.JSON(200, ans)
```

Comment:
I think `res` or `resp` are more idiomatic than `ans`


```suggestion
		resp, err := handler(context)
		if err != nil {
			c.Error(err)
			return
		}

		if c.Request.Header.Get("Accept") == "application/xml" {
			c.XML(200, resp)
			return
		}

		c.JSON(200, resp)
```

For records, this is what I think about when I see `ans` (approximately `hans`)

![n5vsub7rt48a1.jpg](https://github.com/user-attachments/assets/9e3ef25b-32d7-4bb4-be3a-5fc4b04b929d)

Reply:
Yes, fixed!